ra-optimisation/rao-api/src/main/java/com/powsybl/openrao/raoapi/Rao.java

```diff
@@ -86,9 +89,20 @@ public RaoResult run(RaoInput raoInput, RaoParameters parameters, Instant target
                 .findFirst().orElseThrow();
             BUSINESS_WARNS.warn("Running RAO using Open RAO version {} from git commit {}.", openRaoVersion.getMavenProjectVersion(), openRaoVersion.getGitVersion());
 
+            deprecateNonGlobalSecondPreventive(parameters);
+
             return provider.run(raoInput, parameters, targetEndInstant).join();
         }
 
+        public void deprecateNonGlobalSecondPreventive(RaoParameters parameters) {
+            if (parameters.hasExtension(OpenRaoSearchTreeParameters.class)) {
```

Comment:
Use SecondPreventiveRaoPatameters.getSecondPreventiveReOptimizeCurativeRangeActions.
Also change default value for parameter.